apollo-router/src/router.rs

```diff
@@ -279,12 +279,20 @@ impl ConfigurationSource {
     /// Convert this config into a stream regardless of if is static or not. Allows for unified handling later.
     fn into_stream(self) -> impl Stream<Item = Event> {
         match self {
-            ConfigurationSource::Static(instance) => {
+            ConfigurationSource::Static(mut instance) => {
+                if instance.dev.unwrap_or_default() {
```

Comment:
Possibly this logic should be here: https://github.com/apollographql/router/blob/main/apollo-router/src/configuration/mod.rs#L368

@o0Ignition0o WDTY? You were looking at this recently.

Reply:
Hmm still weird to have a mutation in the `validate` method. But I can add this validate call to my tests for dev mode. We will be more safe

Reply:
Yeah, not the `validate` method.
But `ConfigurationSource` still feels like the wrong place. Checking

Reply:
I will open an issue regarding this validation problem. I tried several solutions but each of them has an issue. I need to dig more into it. I will merge this PR to solve the root issue, we also have `validate` called in the tests so we're safe for now but I agree we should find a better way to implement it